trade-finance-manager-api/src/v1/helpers/amendment.helpers.js

```diff
@@ -331,19 +331,21 @@ const firstTaskEmailConfirmation = async (facilityId, amendmentId, auditDetails)
  * @param {string} [deal.dealSnapshot.ukefDealId] - The UKEF deal ID for GEF deals.
  * @param {Object} [deal.dealSnapshot.details] - The details object for BSS deals.
  * @param {string} [deal.dealSnapshot.details.ukefDealId] - The UKEF deal ID for BSS deals.
- * @param {Object} auditDetails - The audit details for logging purposes.
+ * @param {import("@ukef/dtfs2-common").AuditDetails}
  * @returns {Promise<void>} - A promise that resolves when the email has been sent and the flag has been updated.
  */
 const sendFirstTaskEmail = async (deal, auditDetails) => {
   try {
     const { amendment, dealSnapshot, facilityId, amendmentId } = deal;
     const { tasks } = amendment;
-    const { _id: dealId } = deal;
+    const { _id } = deal;
```

Comment:
you could keep this re-assignment to dealId to make it clear, maybe even dealMongoId, but ye good catch the assignment was wrong way round

`const { dealMongoId: _id } = deal;`

Reply:
probably not necessary just thought that's probably where the error is more than anything else! we can keep the logic on line 346 of using the dealSnapshot and just getting whichever one we can if we need

Reply:
Good shout, have moved the logic to LOC:346 should have thought about it.